vulnerabilities/importers/elixir_security.py

```diff
@@ -0,0 +1,120 @@
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import yaml
+import re
+import json
+import requests
+from typing import Set
+from typing import List
+from dephell_specifier import RangeSpecifier
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import GitDataSource
+from vulnerabilities.data_source import GitDataSourceConfiguration
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import Reference
+
+
+class ElixirSecurityDataSource(GitDataSource):
+    def __enter__(self):
+        super(ElixirSecurityDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, file_ext="yml", subdir="./packages"
+            )
+
+    def updated_advisories(self) -> Set[Advisory]:
+        files = self._updated_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    def added_advisories(self) -> Set[Advisory]:
+        files = self._added_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    @staticmethod
+    def generate_all_version_list(pkg_name):
+        resp = requests.get(f"https://hex.pm/api/packages/{pkg_name}")
+        resp = resp.content
+        json_resp = json.loads(resp)
+        version_list = []
+        for release in json_resp["releases"]:
+            version_list.append(release["version"])
+        return version_list
+
+    def get_pkg_from_range(self, version_list, pkg_name):
+        pkg_versions = []
+        all_version_list = self.generate_all_version_list(pkg_name)
+        if version_list is None:
+            return
+        version_ranges = {RangeSpecifier(r) for r in version_list}
+        for version in all_version_list:
+            if any([version in v for v in version_ranges]):
+                pkg_versions.append(version)
+        return pkg_versions
+
+    def process_file(self, path):
+        with open(path) as f:
+            yaml_file = yaml.safe_load(f)
+            pkg_name = yaml_file["package"]
+            safe_pkg_versions = []
+            if yaml_file.get("unaffected_versions"):
+                safe_pkg_versions = self.get_pkg_from_range(
+                    yaml_file["patched_versions"] + yaml_file["unaffected_versions"],
+                    pkg_name,
+                )
+            else:
+                safe_pkg_versions = self.get_pkg_from_range(
+                    yaml_file["patched_versions"], pkg_name
+                )
+            cve_id = yaml_file["cve"]
```

Comment:
This should be 
`cve_id = "CVE-"+yaml_file["cve"] `

Reply:
ok